src/app/services/sms/sms.controller.ts

```diff
@@ -0,0 +1,72 @@
+import { StatusCodes } from 'http-status-codes'
+import { Types } from 'mongoose'
+
+import { ErrorDto } from '../../../types/api'
+import { createLoggerWithLabel } from '../../config/logger'
+import { withUserAuthentication } from '../../modules/auth/auth.middlewares'
+import { ControllerHandler } from '../../modules/core/core.types'
+import * as FormService from '../../modules/form/form.service'
+import * as SmsService from '../../services/sms/sms.service'
+import { createReqMeta } from '../../utils/request'
+
+import { SmsCountsMeta } from './sms.types'
+import { mapRouteError } from './sms.util'
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Private handler to retrieve the free sms counts remaining for a user and a form belonging to the user
+ * This is the controller for GET /:userId/:formId endpoint
+ * @param formId The id of the form to retrieve the message service id for
+ * @returns 200 with msgSrvcId and free sms counts when successful
+ * @returns 404 when the formId is not found in the database
+ * @returns 500 when a database error occurs during retrieval
+ */
+export const _handleGetFreeSmsCountForFormAdmin: ControllerHandler<
+  {
+    formId: string
+  },
+  ErrorDto | SmsCountsMeta
+> = (req, res) => {
+  const { formId } = req.params
+  const logMeta = {
+    action: 'handleGetFreeSmsCountForFormAdmin',
+    ...createReqMeta(req),
+    formId,
+  }
+
+  // Step 1: Check that the form exists
+  return (
+    FormService.retrieveFormById(formId)
+      // Step 2: Retrieve the free sms count
+      .andThen(({ msgSrvcName, admin }) => {
+        // This casting is required because no type parameter is specified on the document.
+        // Hence, this defaults to any but on the schema, this is defined as an ObjectId
+        const adminId = (admin as Types.ObjectId).toHexString()
+        return SmsService.retrieveFreeSmsCounts(adminId).map(
+          (freeSmsCount) => ({
+            msgSrvcSid: msgSrvcName,
+            freeSmsCount,
```

Comment:
nit: should we rename this to `freeSmsCountsRemaining`, this is very vague as to what the count is

Reply:
i shifted the return type for the controller to be in `/api` and renamed this; however, the dto itself has a general naming because it's supposed to just represent a transfer of `smsCount` and what this count specifically refers to should be determined by the caller